### PR TITLE
Restore --debug flag functionality

### PR DESCRIPTION
**Pull request by twz123**
## Description

The debug flag was not functional for the controller, worker and api subcommands since #1333.

Also:

* Remove the explicit dependency to viper.
* Remove ineffective duplicate definition of the debug flag

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

- [x] Manual test
- [ ] Auto test added

## Checklist:

- [x] My code follows the style [guidelines](https://github.com/k0sproject/k0s/blob/main/docs/contributors/overview.md) of this project 
- [x] My commit messages are [signed-off](https://github.com/k0sproject/k0s/blob/main/docs/contributors/github_workflow.md)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: cmd/token/create.go [hunk 1]
```diff
@@ -41,8 +40,6 @@ k0s token create --role worker --expiry 10m  //sets expiration time to 10 minute
 `,
 		PreRunE: checkCreateTokenRole,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			// Disable logrus for token commands
-			logrus.SetLevel(logrus.WarnLevel)
```
Comment (kke) [inline cmd/token/create.go:45]: The token commands could also do `logrus.SetOutput(os.Stderr)`

File: cmd/worker/worker.go [hunk 2]
```diff
@@ -53,10 +53,13 @@ func NewWorkerCmd() *cobra.Command {
 	$ k0s worker --token-file [path_to_file]
 	Note: Token can be passed either as a CLI argument or as a flag`,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			logrus.SetLevel(logrus.InfoLevel)
+			c := CmdOpts(config.GetCmdOpts())
```
Comment (kke) [inline cmd/worker/worker.go:56]: There could be a `c.LogLevel` 🤔

**Review by kke — APPROVED** [review]

File: cmd/token/create.go (hunk 1 above)
Comment (twz123) [reply cmd/token/create.go:45]: stderr/warn is the k0s default, so I think it would be redundant here?

**Comment by twz123** [comment]
@kke would you mind to have another look?

File: cmd/token/create.go (hunk 1 above)
Comment (kke) [reply cmd/token/create.go:45]: Then it shouldn't matter if it outputs warnings as long as the token goes to stdout and everything else to stderr.

**Review by kke — APPROVED** [review]

**Pull request merged**